doc/mobile_base_arm/mobile_base_arm_tutorial.rst

```diff
@@ -0,0 +1,108 @@
+======================================================
+.. image:: images/intro.png
+   :width: 700px
+
+Getting Started
+---------------
+If you haven't already done so, make sure you've completed the steps in `Getting Started <../getting_started/getting_started.html>`_.
+
+Installation
+------------
+
+Install the dependencies: ::
+
+  cd ~/ws_moveit/src
+  wget https://raw.githubusercontent.com/PickNikRobotics/stretch_ros/ros2/stretch_ros.repos
+  vcs import < stretch_ros.repos
+  rosdep install -r --from-paths . --ignore-src --rosdistro foxy -y
+
+Build the colcon workspace: ::
+
+  cd ~/ws_moveit
+  colcon build --event-handlers desktop_notification- status- --cmake-args -DCMAKE_BUILD_TYPE=Release
+
+Source the colcon workspace: ::
+
+  source ~/ws_moveit2/install/setup.bash
+
+Setting up IK solver for mobile-base and arm
+--------------------------------------------
+
+To setup the kinematics plugin we need a joint model group that consist of two joint model groups: one for the manipulator and the other one for the mobile base which has only one joint with planar type so the ``*.srdf`` file should have:
+
+.. code-block:: xml
+
+    <group name="manipulator">
+      ...
+    </group>
+    <group name="mobile_base">
+      <joint name="planar_joint" /> <!-- planar_joint must have planar type -->
+    </group>
+    <group name="mobile_base_manipulator">
+      <group name="manipulator" />
+      <group name="mobile_base" />
+    </group>
+
+In the ``kinematics.yaml`` file the ``mobile_base_manipulator`` group need to have ``stretch_kinematics_plugin/StretchKinematicsPlugin`` as a kinematics solver
+
+.. code-block:: yaml
+
+    mobile_base_manipulator:
+      kinematics_solver: stretch_kinematics_plugin/StretchKinematicsPlugin
+      kinematics_solver_search_resolution: 0.005
+      kinematics_solver_timeout: 0.1
+
+See `stretch_description.srdf <https://github.com/PickNikRobotics/stretch_ros/blob/ros2/stretch_moveit_config/config/stretch_description.srdf#L29-L35>`_ and `kinematics.yaml <https://github.com/PickNikRobotics/stretch_ros/blob/ros2/stretch_moveit_config/config/kinematics.yaml#L5-L8>`_ for how it's been set-up for stretch robot.
+
+For the source-code take a look at `stretch_kinematics_plugin <https://github.com/PickNikRobotics/stretch_moveit_plugins/tree/main/stretch_kinematics_plugin>`_
+
+Setting up the kinematics model for the base
+--------------------------------------------
+
+To be able to generate plan for a differential-drive base we need to set-up two joint properties
+
+motion_model
+++++++++++++
+
+This's used to config the planar joint model, it can have two values:
+
+- holonomic: The joint can move in any direction, this is the default value if no motion_model joint property is specified
+- diff_drive: The joint will have differential drive constraint (can't move side-ways)
+
+.. code-block:: xml
+
+    <joint_property joint_name="planar_joint_name" property_name="motion_model" value="diff_drive" />
```

Comment:
Maybe also document how to set angular weight (which you can now also set via SRDF).

Reply:
I did add it, but I'm not sure why it's being used so I didn't add a description, yet

Reply:
The distance between two base poses (x0,y0,theta0) and (x1,y1,theta1) is the weighted sum of the distance between position and orientation: ||(x0,y0) - (x1,y1)|| - angular_weight*|theta0-theta1|.